src/main/java/fr/xephi/authme/process/join/AsynchronousJoin.java

```diff
@@ -89,8 +89,9 @@ public void processJoin(final Player player) {
         }
 
         if (service.getProperty(RestrictionSettings.FORCE_SURVIVAL_MODE)
-            && !service.hasPermission(player, PlayerStatePermission.BYPASS_FORCE_SURVIVAL)) {
-            bukkitService.runTask(() -> player.setGameMode(GameMode.SURVIVAL));
+            && !service.hasPermission(player, PlayerStatePermission.BYPASS_FORCE_SURVIVAL)
+            && player.getGameMode() != GameMode.SURVIVAL) {
```

Comment:
This is nice—if you want to micro optimize even more I'm guessing checking the game mode is cheaper than checking the permission :wink: